dio/lib/src/adapters/io_adapter.dart

```diff
@@ -115,14 +115,18 @@ class IOHttpClientAdapter implements HttpClientAdapter {
         if (v != null) request.headers.set(k, v);
       });
     } on SocketException catch (e) {
-      if (!e.message.contains('timed out')) {
-        rethrow;
+      if (e.message.contains('timed out')) {
+        throw DioException.connectionTimeout(
```

Comment:
We're now loosing the original stacktrace here. So it should either `Error.throwWithStackTrace()` or the original stacktrace should be passed to the DioException

Reply:
Here we are again 😆  I think the original stacktrace doesn't help the user in figuring out where the problem originated and which Dio call caused it. The stacktrace for the `SocketException` always points to some `dart:io`

Reply:
Let's just pass it anyway since we'll make a huge refactor using `Error.throwWithStackTrace` in future versions. 🤔

Reply:
But then we lose the one that is important.

Reply:
We don't do it for the others (timeout) etc. either.
I created https://github.com/cfug/dio/discussions/1914 to discuss this problem further.